src/main/kotlin/com/exactpro/th2/common/schema/factory/FactorySettings.kt

```diff
@@ -0,0 +1,85 @@
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.group.RabbitMessageGroupBatchRouter
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.parsed.RabbitParsedBatchRouter
+import com.exactpro.th2.common.schema.message.impl.rabbitmq.raw.RabbitRawBatchRouter
+import java.nio.file.Path
+
+
+data class FactorySettings @JvmOverloads constructor(
+    var messageRouterParsedBatchClass: Class<out MessageRouter<MessageBatch>> = RabbitParsedBatchRouter::class.java,
+    var messageRouterRawBatchClass: Class<out MessageRouter<RawMessageBatch>> = RabbitRawBatchRouter::class.java,
+    var messageRouterMessageGroupBatchClass: Class<out MessageRouter<MessageGroupBatch>> = RabbitMessageGroupBatchRouter::class.java,
+    var eventBatchRouterClass: Class<out MessageRouter<EventBatch>> = EventBatchRouter::class.java,
+    var grpcRouterClass: Class<out GrpcRouter> = DefaultGrpcRouter::class.java,
+    var rabbitMQ: Path? = null,
+    var routerMQ: Path? = null,
+    var connectionManagerSettings: Path? = null,
+    var grpc: Path? = null,
+    var routerGRPC: Path? = null,
+    var cradleConfidential: Path? = null,
+    var cradleNonConfidential: Path? = null,
+    var prometheus: Path? = null,
+    var boxConfiguration: Path? = null,
+    private var _custom: Path? = null,
+    private var _dictionariesDir: Path? = null,
+    var variables: Map<String, String> = emptyMap()) {
+
+    var custom: Path?
+    get() = _custom ?: CommonFactory.CONFIG_DEFAULT_PATH.resolve(CommonFactory.CUSTOM_FILE_NAME)
+    set(value) { _custom = value }
+
+    var dictionariesDir: Path?
+    get() = _dictionariesDir ?: CommonFactory.CONFIG_DEFAULT_PATH
+    set(value) {_dictionariesDir = value}
+
+    fun createConfigurationManager(): ConfigurationManager = ConfigurationManager(HashMap<Class<*>, Path>().also { paths ->
+        paths[RabbitMQConfiguration::class.java] = defaultPathIfNull(rabbitMQ, CommonFactory.RABBIT_MQ_FILE_NAME)
+        paths[MessageRouterConfiguration::class.java] = defaultPathIfNull(routerMQ, CommonFactory.ROUTER_MQ_FILE_NAME)
+        paths[ConnectionManagerConfiguration::class.java] = defaultPathIfNull(connectionManagerSettings, CommonFactory.CONNECTION_MANAGER_CONF_FILE_NAME)
+        paths[GrpcConfiguration::class.java] = defaultPathIfNull(grpc, CommonFactory.GRPC_FILE_NAME)
+        paths[GrpcRouterConfiguration::class.java] = defaultPathIfNull(routerGRPC, CommonFactory.ROUTER_GRPC_FILE_NAME)
+        paths[CradleConfidentialConfiguration::class.java] = defaultPathIfNull(cradleConfidential, CommonFactory.CRADLE_CONFIDENTIAL_FILE_NAME)
+        paths[CradleNonConfidentialConfiguration::class.java] = defaultPathIfNull(cradleNonConfidential, CommonFactory.CRADLE_NON_CONFIDENTIAL_FILE_NAME)
+        paths[PrometheusConfiguration::class.java] = defaultPathIfNull(prometheus, CommonFactory.PROMETHEUS_FILE_NAME)
+        paths[BoxConfiguration::class.java] = defaultPathIfNull(boxConfiguration, CommonFactory.BOX_FILE_NAME)
+    })
```

Comment:
Why the settings class creates the configuration manager? Probably, we should have the factory method in the ConfigurationManager to create the manager using FactorySettings